src/registrar/management/commands/disclose_security_emails.py

```diff
@@ -0,0 +1,77 @@
+""""
+Converts all ready and DNS needed domains with a non-default public contact
+to disclose their public contact. Created for Issue#1535 to resolve
+ disclose issue of domains with missing security emails.
+"""
+
+import logging
+import copy
+
+from django.core.management import BaseCommand
+from registrar.models import Domain
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Disclose all nondefault domain security emails."
+
+    def __init__(self):
+        """Sets global variables for code tidyness"""
+        super().__init__()
+        # domains and transition domains that must be disclosed to true
+        self.contacts_saved_count = 0
+        # domains with errors, which are not successfully updated to disclose
+        self.domains_with_errors: list[str] = []
+        # domains that are successfully disclosed
+        self.disclosed_domain_contacts_count = 0
+        # domains that skip disclose due to having contact registrar@dotgov.gov
+        self.skipped_domain_contacts_count = 0
+
+    def handle(self, **options):
+        """
+        Converts all ready and DNS needed domains with a non-default public contact
+        to disclose their public contact.
+        """
+        logger.info("Updating security emails to public")
+
+        # Initializes domains that need to be disclosed
+
+        statuses = [Domain.State.READY, Domain.State.DNS_NEEDED]
+        domains = Domain.objects.filter(state__in=statuses)
+
+        logger.info(f"Found {len(domains)} domains with status Ready or DNS Needed.")
+
+        # Call security_contact on all domains to trigger saving contact information
```

Comment:
Why are we doing this here? We then have to have two for loops over 28000ish records. if we just want a count there is domains.count() feature will return the count. or even doing len(domains) so we don't have to iterate over them multiple times.

Reply:
My assumption was that we'd have to first call security_contact to properly update all the contacts before updating the rest of the domains. If that isn't the case and we can call security_contact and disclose on relevant domains in the same for loop, I can merge the two

Reply:
Just pushed a change merging the two loops -- feel free to let me know if this isn't the suggested change that you had in mind